client/navigation/components/container/index.js

```diff
@@ -22,6 +22,68 @@ import CategoryTitle from '../category-title';
 import Header from '../header';
 import Item from '../../components/Item';
 
+/**
+ * Get a map of all categories, including the topmost WooCommerce parentCategory
+ *
+ * @param {Array} menuItems Array of menuItems
+ * @return {Object} Map of categories by id
+ */
+export const getCategoriesMap = ( menuItems ) => {
```

Comment:
Optional, but would be nice to move these 2 functions to a utils file to reduce some of the noise in this component.